### DISCUSSION TITLE
S3 policy for quota

### DISCUSSION THREAD

**Discussion by aoberest** [seaweedfs/seaweedfs#2024] [Q&A] [answered]
Hello.
Is it possible to restrict recording into backet when the quota is exceeded?

**Comment by chrislusf** [comment, marked as answer]
Currently there are no quota controls implemented.

I am planning to add an enterprise/paid add-on component to manage quota. Let me know if you are interested.

**Reply by chrislusf** [reply]
The cloud monitoring tool has the option to set quota now. https://github.com/chrislusf/seaweedfs/wiki/Cloud-Monitoring

**Reply by aoberest** [reply]
Thanks for the new functionality
But we use seaweedfs in a closed loop and we don't have internet access.

**Comment by IcedQuinn** [comment]
It ought to be possible to do something like this with a simple proxy and a postgres database. Query the database for that user to sum the bytes and check if that goes over quota. Otherwise proxy the upload to seaweed and replace insert the used size for that file id in the database.

**Reply by chrislusf** [reply]
counting the bytes would cost a lot of time.

**Reply by IcedQuinn** [reply]
> counting the bytes would cost a lot of time.

you tend to get the size of a payload via `Content-Length` or incrementally for chunked encoding. if one is doing their own quota system at the API level then you'd sum up the length counters while uploading and write a map of the file ID to size in postgres. plus you can view chunk IDs that belong to particular users which may be a GDPR requirement.
